### ISSUE TITLE
Unit tests: get_cloud(distro=FreeBSD) returns a bogus OS Family 

### ISSUE THREAD

**Issue by igalic** [canonical/cloud-init#4118]
```python
In [1]: from tests.unittests.util import get_cloud

In [2]: from unittest import mock

In [3]:         with mock.patch(
   ...:             "cloudinit.distros.networking.subp.subp", return_values=("", None)
   ...:         ):
   ...:             cloud = get_cloud(distro="freebsd", metadata={})
   ...: 

In [4]: cloud.distro.name
Out[4]: 'freebsd'

In [5]: cloud.distro.osfamily
Out[5]: 'linux'

In [6]: 
```

**Comment by TheRealFalcon** [comment]
@igalic want me to assign you to this one?

**Comment by igalic** [comment]
not sure i can figure this out without handholding
I'm finding our test framework very confusing

**Comment by blackboxsw** [comment]
I think the issue is  [cloudinicloudinit/distros/bsd.py:        `self.osfamily = platform.system().lower()`](https://github.com/canonical/cloud-init/blob/main/cloudinit/distros/bsd.py#L38).  The tests.unittests.get_cloud helper is leaking through to `cloudinit.util.platform`  on the underlying host system, but I think you want `def get_cloud` to either mock `platform.system()` 


might think about something like:
```diff

diff --git a/tests/unittests/util.py b/tests/unittests/util.py
index 8c6682e18..e8c5c90e5 100644
--- a/tests/unittests/util.py
+++ b/tests/unittests/util.py
@@ -20,8 +20,11 @@ def get_cloud(
     """
     paths = paths or helpers.Paths({})
     sys_cfg = sys_cfg or {}
+    import pdb; pdb.set_trace()
     cls = distros.fetch(distro) if distro else MockDistro
-    mydist = cls(distro, sys_cfg, paths)
+    # *BSD calls platform.system to determine osfamilies
+    with mock.patch('platform.system', return_value=distro.lower()):
+        mydist = cls(distro, sys_cfg, paths)
     if mocked_distro:
         mydist = mock.MagicMock(wraps=mydist)
     myds = DataSourceTesting(sys_cfg, mydist, paths)
```

**Comment by TheRealFalcon** [comment]
Fixed in #4119
